test/core/channel/metrics_test.cc

```diff
@@ -646,6 +646,19 @@ TEST_F(MetricsDeathTest, RegisterTheSameMetricNameWouldCrash) {
       "Metric name double_histogram has already been registered.");
 }
 
+// Test that we can use the target and authority after the original strings are
+// freed.
+TEST(StatsPluginChannelScopeTest, UseAfterFree) {
```

Comment:
This is kind of too low-level of a test.  What I was really asking for was a test that demonstrates the crash that prompted this fix.  In what circumstance does that happen?

Actually, this makes me realize that I really don't understand why this fix is necessary in the first place.  The ChannelScope shouldn't be used asynchronously, so the underlying strings should always exist until after they're accessed here.  In what situation is that not happening?

Reply:
https://github.com/grpc/grpc/blob/fb6a57b153c1290c08028cd909d759fcd393ff22/src/core/lib/surface/legacy_channel.cc#L96 for example. The authority is a temporary string and also where the issue exists. I could also change the invocation to make sure that `GetStatsPluginForChannel` is invoked in the same scope as target and authority, but that seems too brittle.

Reply:
I think we should absolutely fix that caller to keep the string in scope instead of changing `ChannelScope` to store copies of the strings.  We should not be using `ChannelScope` in a situation where it needs its own copy.

Reply:
I still think that this is too brittle, but the test would atleast make sure that target and authority are accessible.